extensions/ql-vscode/src/data-extensions-editor/queries/java.ts

```diff
@@ -148,6 +146,82 @@ boolean isSupported(CallableMethod method) {
   or
   not method.isSupported() and result = false
 }
+
+// The below is a copy of https://github.com/github/codeql/blob/249f9f863db1e94e3c46ca85b49fb0ec32f8ca92/java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll
+// to avoid the use of internal modules.
+/** Holds if the given package \`p\` is a test package. */
+pragma[nomagic]
+private predicate isTestPackage(Package p) {
+  p.getName()
+      .matches([
+          "org.junit%", "junit.%", "org.mockito%", "org.assertj%",
+          "com.github.tomakehurst.wiremock%", "org.hamcrest%", "org.springframework.test.%",
+          "org.springframework.mock.%", "org.springframework.boot.test.%", "reactor.test%",
+          "org.xmlunit%", "org.testcontainers.%", "org.opentest4j%", "org.mockserver%",
+          "org.powermock%", "org.skyscreamer.jsonassert%", "org.rnorth.visibleassertions",
+          "org.openqa.selenium%", "com.gargoylesoftware.htmlunit%", "org.jboss.arquillian.testng%",
+          "org.testng%"
```

Comment:
This seems quite tedious to maintain separately. I am ok with it for now, but we should open an issue to keep track of this. Ideally this is fixed by moving to the CodeML queries.

Reply:
This will be part of moving to the CodeML queries, they already [have this implemented for their framework mode](https://github.com/github/codeql/blob/2c99009c1ab6692296460afb38d185ae8d2b10f1/java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll#L196-L206), so I don't think we need to open a separate issue to keep track of this.